File: domain/src/main/kotlin/com/kyovo/domain/port/primary/AuthUseCase.kt
package com.kyovo.domain.port.primary

import com.kyovo.domain.model.user.NewUser
import com.kyovo.domain.model.user.User
import com.kyovo.domain.model.user.UserEmail

interface AuthUseCase
{
    fun register(newUser: NewUser): User

    fun login(email: UserEmail, rawPassword: String): User
}

File: frontend/src/domain/port/primary/AuthUseCase.ts
import type { User } from '@/domain/model/User'

export interface Credentials {
  email: string
  password: string
}

export interface Registration {
  name: string
  email: string
  password: string
}

export interface AuthUseCase {
  login(credentials: Credentials): Promise<{ user: User; token: string }>
  register(registration: Registration): Promise<User>
}
